src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -773,8 +773,10 @@ void FlexPA::prepPoint_pin_checkPoint_planar(
       ep, layerPolys, bp, ap->getLayerNum(), dir);
   // skip if two width within shape for standard cell
   if (!isOutSide) {
-    ap->setAccess(dir, false);
-    return;
+        if (!instTerm || !instTerm->getInst()->getMaster()->getMasterType().isBlock()) {
+            ap->setAccess(dir, false);
+            return;
+        }
   }
```

Comment:
I'm not sure I understand how this is supposed to work.  If the ep is not outside the pin won't the gc test be a no-op?  Does this change rely on your other pathseg for ap change?

Reply:
It is still relevant to do DRC because the begin point may be too close to the pin border an thus the path seg edge may be outside. No, this is separate from the APs with path segs feature.

Reply:
If the bp & ep are entirely within the pin shape then nothing is being checked.

Reply:
I'm thinking of:
![image](https://user-images.githubusercontent.com/761514/162263453-b7968a3e-ae40-419c-9439-50a677c94fc6.png)

where this would give south access even though it is really blocked.

Reply:
> If the bp & ep are entirely within the pin shape then nothing is being checked.

what piece of code tells this?

Reply:
isOutSide

Reply:
> I'm thinking of: ![image](https://user-images.githubusercontent.com/761514/162263453-b7968a3e-ae40-419c-9439-50a677c94fc6.png)
> 
> where this would give south access even though it is really blocked.

if there is a blockage there, then there are fixed shape cost there two, so the path wont freely go through the blockage. It is out of scope of the access directions to look so further away.

Reply:
Currently pin access doesn't allow an AP as above because it is checking for this case.  You are disabling that check.  It does expect to find other aps to check where the stub would stick out and check for this.  I believe you have the case where no other ap can produce a proper check.

You had suggested

- Extend the checking pathseg in macro pin cases
- Look at lower tracks in addition of upper tracks for planar access only. 

but this approach seems to be neither of these.

Reply:
I understand that the reason for the wire stub check is not to check if there are blockages nearby because this will be caught by the path search, but to (1) discourage routing through the pin (which potentially can cause DRVs, but not in a single-rectangle pin) and (2) catch DRVs from the connection of the wire stub into the pin, which I considered that couldnt happen with a single rectangle pin, but now it comes to me that there may be a minstep in a pin edge.

Yes the implemented code was neither of these. I was going to do the first one but it didn't make much sense after all. The idea was to allow access to an ap that is far from the border, so in this case I would have to extend the ep to outside of the border and even consider minstep to avoid this violation. And considering my assumptions, all this was unnecessary since you can just allow the access inside the pin.
But I think now it should be extended beyond the pin, perhaps 1 track long or to the next track?

Reply:
It is to check if there is a blockage nearby to prevent generating false aps.  These can block the generation of real aps and can confuse the global router (we had a case with a long pin and many false aps which made the router pick the wrong gcell for the pin when using pin_access).

As for the extension I think at least one track beyond the pin bound is reasonable.

Reply:
I did the change and started a new secure-CI.